sklearn/tests/random_seed.py

```diff
@@ -3,96 +3,107 @@
 from random import Random
 from datetime import datetime
 
-# Definition of the random_seed fixture. See the docstring of the fixture for
-# more information.
-
-RANDOM_SEED_RANGE = list(range(100))  # All seeds in [0, 99] should be valid.
-random_seed_var = environ.get("SKLEARN_TESTS_GLOBAL_RANDOM_SEED")
-if random_seed_var is None:
-    # This is the way.
-    random_seeds = [42]
-elif random_seed_var == "any":
-    # Pick-up one seed at random in the range of admissible random seeds.
-    #
-    # Note, to make sure that all pytest-xdist workers see the same seed, we
-    # seed the meta-random number generator with a value derived either from a
-    # CI build number variables, or from the year and the day for local builds
-    try:
-        meta_seed = int(environ["BUILD_NUMBER"])
-    except (KeyError, ValueError):
+
+# Passes the main worker's random seeds to workers
+class XDistHooks:
+    def pytest_configure_node(self, node) -> None:
+        random_seeds = node.config.getoption("random_seeds")
+        node.workerinput["random_seeds"] = random_seeds
+
+
+def pytest_configure(config):
+    if config.pluginmanager.hasplugin("xdist"):
+        config.pluginmanager.register(XDistHooks())
+
+    # Definition of the random_seed fixture. See the docstring of the fixture for
+    # more information.
+    RANDOM_SEED_RANGE = list(range(100))  # All seeds in [0, 99] should be valid.
+    random_seed_var = environ.get("SKLEARN_TESTS_GLOBAL_RANDOM_SEED")
+    if hasattr(config, "workinput"):
+        # Set worker random seed from seed generated from main process
+        random_seeds = config.workerinput["random_seeds"]
+    elif random_seed_var is None:
+        # This is the way.
+        random_seeds = [42]
+    elif random_seed_var == "any":
+        # Pick-up one seed at random in the range of admissible random seeds.
```

Comment:
Since this is consistent now for xdist, maybe we can actually do "any" and not use the datetime as a seed.